spec/models/concerns/digital_object/datacite_spec.rb

```diff
@@ -0,0 +1,95 @@
+require 'rails_helper'
+require 'equivalent-xml'
+
+describe DigitalObject::Datacite do
+
+  let(:test_class) do
+    _c = Class.new
+    _c.send :include, DigitalObject::Datacite
+  end
+
+  let(:digital_object) do
+    test_do = test_class.new 
+    data = JSON.parse( fixture('lib/hyacinth/ezid/ezid_item.json').read )
+    expect(data).to have_key('dynamic_field_data')
+    data['identifiers'] = ['item.' + SecureRandom.uuid] # random identifer to avoid collisions
+    test_do.instance_variable_set(:@digital_object_data, data)
+    allow(test_do).to receive(:as_json).and_return(data)
+    test_do.instance_variable_set(:@fedora_object,
+                              ContentAggregator.new(pid: 'test:existingobject'))
+    test_do
+  end
+
+  context "#mint_and_store_doi:" do
+    it "mint_and_store_doi" do
+      # stub out method wrapping API call to EZID server, stub return Doi instance
+      allow_any_instance_of(Hyacinth::Datacite::ApiSession).to receive(:mint_identifier) do
+        Hyacinth::Datacite::Doi.new('10.33555/5x55-t644')
+      end
+      EZID[:user] = EZID[:ezid_test_user]
```

Comment:
You should use `stub_const` instead of modifying the constant because I believe direct changes to this constant will leak into other tests.

https://relishapp.com/rspec/rspec-mocks/docs/mutating-constants

Reply:
Since the underlying code reads directly from the EZID hash, stub_const won't work without changes to the underlying code.
Furthermore, the original code, spec/models/concerns/digital_object/ezid_spec.rb, modifies the constant in the same way. Since one of the requirements of this implementation was to change as little code as possible, I suggest we leave it as it is.

Reply:
@fcd1 If you want to use the "real" EZID hash as the source data for stub_const (and then merge in any test-specific changes) then I think it should still work.

I understand that you're trying to change as little source implementation code as possible.  In the above comment, I wasn't asking you to change the underlying code though, just the test.  Even if you'd prefer not to make the change, I hope my comment intention was clear/helpful?